### PR TITLE
fix: remove legacy code

### PR DESCRIPTION
**Pull request by ra-phael**
- **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Remove unused v2 and mixed route logic and legacy router
- **What is the current behavior?** (You can also link to an open issue here)

- **What is the new behavior (if this is a feature change)?**

- **Other information**:

### CODE REVIEW

File: cli/base-command.ts [hunk 1]
```diff
@@ -263,80 +255,66 @@ export abstract class BaseCommand extends Command {
...
-      const gasPriceCache = new NodeJSCache<GasPrice>(
-        new NodeCache({ stdTTL: 15, useClones: true })
-      );
+    const gasPriceCache = new NodeJSCache<GasPrice>(
+      new NodeCache({ stdTTL: 15, useClones: true })
+    );
 
-      // const useDefaultQuoteProvider =
-      //   chainId != ChainId.ARBITRUM_ONE && chainId != ChainId.ARBITRUM_RINKEBY;
+    // const useDefaultQuoteProvider =
```
Comment (0xpApaSmURf) [inline cli/base-command.ts:262]: Can be deleted?

File: src/routers/alpha-router/gas-models/gas-model.ts [hunk 2]
```diff
@@ -94,7 +94,7 @@ export type BuildOnChainGasModelFactoryType = {
   gasPriceWei: BigNumber;
   v3poolProvider: IV3PoolProvider;
   token: Token;
-  v2poolProvider: IV2PoolProvider;
+  v2poolProvider?: IV2PoolProvider;
```
Comment (0xpApaSmURf) [inline src/routers/alpha-router/gas-models/gas-model.ts:97]: Is this removable?

File: cli/base-command.ts (hunk 1 above)
Comment (ra-phael) [reply cli/base-command.ts]: I guess

File: src/routers/alpha-router/gas-models/gas-model.ts (hunk 2 above)
Comment (ra-phael) [reply src/routers/alpha-router/gas-models/gas-model.ts]: Removed in https://github.com/violetprotocol/mauve-smart-order-router/pull/32/commits/ad2362124cee95942aa53006fcc010cd8858c43f

File: cli/base-command.ts (hunk 1 above)
Comment (ra-phael) [reply cli/base-command.ts]: https://github.com/violetprotocol/mauve-smart-order-router/pull/32/commits/844226c47e63a5ac8a77945a44478795f32f915d

**Pull request merged**